java/vector/src/main/java/org/apache/arrow/vector/ipc/WriteChannel.java

```diff
@@ -73,7 +73,9 @@ public long align() throws IOException {
   public long write(ByteBuffer buffer) throws IOException {
     long length = buffer.remaining();
     LOGGER.debug("Writing buffer with size: " + length);
-    out.write(buffer);
+    while(buffer.remaining() > 0) {
```

Comment:
would you mind changing this `while (buffer.hasRemaining()` to simplify a bit?

Reply:
Done